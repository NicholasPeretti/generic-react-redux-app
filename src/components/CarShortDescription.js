import React from 'react'
import PropTypes from 'prop-types'

var CarShortDescription = ({ stockNumber, mileage = {}, color, fuelType }) => (
  <p>
    Stock # {stockNumber} - {mileage.number} {mileage.unit} - {fuelType} - {color}
  </p>
)

CarShortDescription.propTypes = {
  stockNumber: PropTypes.number.isRequired,
  mileage: PropTypes.shape({
    number: PropTypes.number.isRequired,
    unit: PropTypes.string.isRequired
  }).isRequired,
  color: PropTypes.string.isRequired,
  fuelType: PropTypes.string.isRequired
}

export default CarShortDescription
