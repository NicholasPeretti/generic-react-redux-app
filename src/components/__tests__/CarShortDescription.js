import React from 'react'
import { shallow } from 'enzyme'
import PresentationalTest from '../../utils/tests/presentational'
import CarShortDescription from '../CarShortDescription'
const props = {
  stockNumber: 12345,
  mileage: {
    number: 123,
    unit: 'km'
  },
  color: 'white',
  fuelType: 'diesel'
}

describe('CarShortDescription', () => {
  PresentationalTest(CarShortDescription, props)

  const wrapper = shallow(<CarShortDescription {...props}/>)

  test('Should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
