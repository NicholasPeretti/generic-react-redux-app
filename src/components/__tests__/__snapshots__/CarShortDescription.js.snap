// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarShortDescription Should match snapshot 1`] = `"<p>Stock # 12345 - 123 km - diesel - white</p>"`;
